src/views/jobLog/log.ts

```diff
@@ -14,24 +14,37 @@ import { ContextValue } from "../../ibmiProjectExplorer";
 export default class Log extends TreeItem implements ProjectExplorerTreeItem {
   static contextValue = ContextValue.log;
   jobLogInfo: JobLogInfo;
+  showFailedCommands : boolean;
 
   constructor(public workspaceFolder: WorkspaceFolder, jobLogInfo: JobLogInfo, isLocal: boolean = false) {
     super(jobLogInfo.createdTime.toLocaleString(), TreeItemCollapsibleState.Collapsed);
 
     this.jobLogInfo = jobLogInfo;
     this.iconPath = new ThemeIcon('archive', isLocal ? new ThemeColor('joblog.local') : undefined);
     this.contextValue = Log.contextValue;
+    this.showFailedCommands = false;
+  }
+
+  toggleShowFailed(): void {
+    this.showFailedCommands = this.showFailedCommands === false ? true : false;
   }
 
   getChildren(): ProjectExplorerTreeItem[] {
     let items: ProjectExplorerTreeItem[] = [];
 
     const jobLogInfo = this.jobLogInfo;
 
-    items.push(...jobLogInfo.commands?.map(
-      command => new Command(this.workspaceFolder, command)
-    ));
-
+    if (this.showFailedCommands) {
+      items.push(...jobLogInfo.commands?.filter(
+        command => command.failed === this.showFailedCommands
```

Comment:
you already know the `showFailedCommands` is `true`
so why not just have `command.failed`